capstone/src/main/java/com/google/step/servlets/LoginServlet.java

```diff
@@ -0,0 +1,103 @@
+import com.google.appengine.api.datastoreDatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserserviceFactory();
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.util.Map;
+import java.util.HashMap;
+import com.google.gson.Gson;
+
+/**
+ * Servlet that will handle login information.
+ */
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
```

Comment:
Just so I understand conceptually, will this servlet be called onload in the body of the landing page? Then once we get the json back on the frontend, we redirect accordingly?

Reply:
Yes, it's called onload in the body of the landing page.

No, so what happens is the following: within this servlet, I create 3 login links, 1 if the user wants to simply login (which will then redirect to index.html,2 if the user wants to sign up as a user (which will then redirect to the user-signup.html (fake name created for now)), 3 if the restaurant wants to sign up as a restaurant (which will then redirect to restaurant-signup.html). In these sign up html's is where the user or restaurant will input their data so that we then store it into datastore. Those will make a POST into the other 2 servlets and they redirect back to the index.html.